src/mrpro/data/_KTrajectoryRawShape.py

```diff
@@ -0,0 +1,80 @@
+"""KTrajectoryRawShape dataclass."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+import numpy as np
+import torch
+from einops import rearrange
+
+from mrpro.data import KTrajectory
+
+
+@dataclass(slots=True, init=False)
+class KTrajectoryRawShape:
+    """K-space trajectory in the shape of ((other*k2*k1),k0).
+
+    Order of directions is always kz, ky, kx
+    Shape of each of kx,ky,kz is ((other,k2,k1),k0) this means that e.g. slices, averages... have not yet been
+    separated from the phase and slice encoding dimensions. The trajectory is in the same shape as the raw data in the
+    raw data file.
+    """
+
+    kz: torch.Tensor  # ((other,k2,k1),k0) #phase encoding direction, k2 if Cartesian
+    ky: torch.Tensor  # ((other,k2,k1),k0) #phase encoding direction, k1 if Cartesian
+    kx: torch.Tensor  # ((other,k2,k1),k0) #frequency encoding direction, k0 if Cartesian
+
+    def __init__(self, kz: torch.Tensor, ky: torch.Tensor, kx: torch.Tensor):
```

Comment:
I think we dont need the init here?
Can we just use default dataclass init?
I.e.  `@dataclass(slots=True)` and no init